src/renderers/shared/stack/reconciler/ReactChildReconciler.js

```diff
@@ -27,8 +28,9 @@ function instantiateChild(childInstances, child, name) {
       keyUnique,
       'flattenChildren(...): Encountered two children with the same key, ' +
       '`%s`. Child keys must be unique; when two children share a key, only ' +
-      'the first child will be used.',
-      KeyEscapeUtils.unescape(name)
+      'the first child will be used.%s',
+      KeyEscapeUtils.unescape(name),
+      ReactComponentTreeDevtool.getCurrentStackAddendum(childInstances[name])
```

Comment:
This is not a React element; it is an internal React component instance. This is why you got the useless "in Unknown" stack frame. I think `childInstances[name]._currentElement` is pretty close to what you want here, but that will actually not give you a useful stack if `childInstances[name]` is a text component. This error message could also be misleading if the duplicate keys are on components of different types since you'd only see one of them.

Instead, I think it probably makes sense to only show the stack to the parent component – but it is actually going to be a bit complicated to thread that information through. getCurrentStackAddendum always looks at the *current* owner (hence its name), but during reconciliation here we shouldn't rely on the owner being set. I'm actually surprised that it even is in your test case. Let's add a new function ReactComponentTreeDevtool.getStackAddendum(id) that takes a debug ID and prints the parents of that instance. We'll want to reuse some of the code from getCurrentStackAddendum but nothing about topElement or ReactCurrentOwner.

You'll probably want to add a parameter to ReactChildReconciler.instantiateChildren that takes the debug ID of the parent component, then pass that from ReactMultiChild.